ntbs-service/Services/AdImportService.cs

```diff
@@ -31,22 +31,22 @@ public async Task RunCaseManagerImportAsync()
             using (var adDirectoryService = _adDirectoryServiceFactory.Create())
             {
                 var users = adDirectoryService.LookupUsers(tbServices).ToList();
+                var ntbsUsersNotInAd = this._userRepository.GetUserQueryable()
+                    .Where(user => !users.Select(u => u.user.Username).Contains(user.Username)).ToList();
+                var ntbsUsersNotInAdWithTbServices = ntbsUsersNotInAd.Select(u =>
+                    (u, u.CaseManagerTbServices.Select(cmtb => cmtb.TbService).ToList()));
+                users.AddRange(ntbsUsersNotInAdWithTbServices);
                 foreach (var (user, tbServicesMatchingGroups) in users)
                 {
                     Log.Information($"Updating user {user.Username}");
+                    if (ntbsUsersNotInAd.Select(u => u.Username).Contains(user.Username))
+                    {
+                        user.IsActive = false;
+                        user.AdGroups = null;
+                    }
+
                     await _userRepository.AddOrUpdateUser(user, tbServicesMatchingGroups);
                 }
-                
-                var ntbsUsersNotInAd = this._userRepository.GetUserIQueryable()
-                    .Where(user => !users.Select(u => u.user.Username)
-                        .Contains(user.Username)).ToList();
-                foreach (var user in ntbsUsersNotInAd)
-                {
-                    Log.Information($"Updating user {user.Username}");
-                    user.IsActive = false;
-                    user.AdGroups = null;
-                    await _userRepository.AddOrUpdateUser(user, user.CaseManagerTbServices.Select(cmtb => cmtb.TbService));
-                }

```

Comment:
Sorry, entirely my fault, but I don't think this is a good change. I misread what it was doing before - I hadn't noticed that the second set of users is entirely distinct from the first one. I think we should revert this.

Reply:
However, I do still think we should commonise this logic

Reply:
I have now moved the logic into the new AdUserService, and reverted back to the separate foreach loops.